src/components/Support.js

```diff
@@ -2,9 +2,15 @@ import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { Button, Card, CardContent, Grid, Typography } from "@material-ui/core";
 import { makeStyles } from "@material-ui/core/styles";
 import clsx from "clsx";
+import dynamic from "next/dynamic";
 import PropTypes from "prop-types";
 import React from "react";
-import { TwitterShareButton } from "react-share";
+// import { TwitterShareButton } from "react-share";
+
+const TwitterShareButton = dynamic(
+  () => import("react-share").then((mod) => mod.TwitterShareButton),
+  { ssr: false },
+);
```

Comment:
What is happening/not happening here?

Reply:
We're using this same component in almost all other ups e.g. charterafrica, codeforafrica, etc., and I don't think we've ever had to load it dynamically.

It could be we're using some outdated version in this repo.

Reply:
@kilemensi  from the tests we did we found out that the hydration error occurs because there is a button nested within the `TwitterShareButton` as shown below👇

```
    <TwitterShareButton
            url=""
            title=""
            via=""
            hashtags={[""]}
          >
            <Button
              variant="contained"
              color="primary"
              className={classes.button}
            >
              <FontAwesomeIcon
                className={classes.fa}
                icon={["fab", "twitter"]}
                size="2x"
                fixedWidth
              />{" "}
              TWEET
            </Button>
          </TwitterShareButton>
```

If we remove the child button, it works without any errors but we would lose the MIU styles, as of now we were not able to cleanly replicate the MUI button style on the `TwitterShareButton`  but would've been the best option

Reply:
If I got you, then `Button` should be the main component and `TwitterShareButton` should be set on `component` prop of the `Button`; similar-ish to how it's one in [`codeforafrica`](https://github.com/CodeForAfrica/ui/blob/main/apps/codeforafrica/src/components/ShareButton/ShareButton.js) app.

Reply:
Yeah this could work as well... I had a different idea 🤞  `TwitterShareButton` can still remain as the main component if we change the `component`  prop of `Button` to something else like `Box` that contains the Icon and Text would also resolve the hydration issues. I will touch base with @thepsalmist on the proposed solutions. 

```
      <TwitterShareButton
            url="https://sensors.AFRICA/air"
            title="Did you know the #AirQuality in our city directly affects my health? Check our city’s pollution levels on"
            via="sensorsAFRICA"
            hashtags={["sensorsAFRICA"]}
          >
            <Button
              variant="contained"
              color="primary"
              className={classes.button}
              component="box"
            >
              <FontAwesomeIcon
                className={classes.fa}
                icon={["fab", "twitter"]}
                size="2x"
                fixedWidth
              />
              TWEET
            </Button>
          </TwitterShareButton>
```

Reply:
Cool, you guy choose whichever method works without dynamic loading.

For added context, the reason of suggestion `Button` as main component is also to reduce the number of elements generated. If `TwitterShareButton` is the main component, you'll end up with two elements e.g. `<button><button>...` where the first `<button>` is whatever `TwitterShareButton` renders to.

If you make `Button` the main component, a single element `<button>` will be rendered.